scripts/ci/pack_soljson.sh

```diff
@@ -0,0 +1,46 @@
+#!/bin/bash
+set -euo pipefail
+
+SCRIPT_DIR="$(realpath "$(dirname "$0")")"
+SOLJSON_JS="$1"
+SOLJSON_WASM="$2"
+SOLJSON_WASM_SIZE=$(wc -c "${SOLJSON_WASM}" | cut -d ' ' -f 1)
+OUTPUT="$3"
+
+(( $# == 3 )) || { >&2 echo "Usage: $0 soljson.js soljson.wasm packed_soljson.js"; exit 1; }
+
+# If this changes in an emscripten update, it's probably nothing to worry about,
+# but we should double-check when it happens and adjust the tail command below.
+[ "$(head -c 5 "${SOLJSON_JS}")" == "null;" ] || { >&2 echo 'Expected soljson.js to start with "null;"'; exit 1; }
+
+echo "Packing $SOLJSON_JS and $SOLJSON_WASM to $OUTPUT."
+(
+	echo -n 'var Module = Module || {}; Module["wasmBinary"] = '
```

Comment:
By the way: Adding the binary to ``Module["wasmBinary"]`` seems to be an official, emscripten-supported way to prefetch the binary, so this should always work - however there is one caveat:
I think this will mean that the binary data will remain live in memory in the module throughout the lifetime of the module...
Not sure we should be bothered by that - 20MB of memory are usually insignificant these days.

We could probably avoid this by injecting the binary more locally into emscriptens wrapper logic, but then we're much more likely to hit issues, if emscripten ever changes the wrapper in an update...

A compromise may be to append ``Module["wasmBinary"] = undefined;`` to the very end of ``soljson.js``, I think that should work and allow the binary to be garbage-collected...

Reply:
> I think this will mean that the binary data will remain live in memory in the module throughout the lifetime of the module...

Isn't this already the case? At least the OOM errors I just saw in https://github.com/ConsenSys/solc-typed-ast/pull/92 would suggest that.

Reply:
It needs to be there right away for the wasm compilation to kick off. But we don't need it afterwards, so I'm just resetting the field and see if that works.

Reply:
Looking at the emscripten wrapper: the local ``wasmBinary`` will be assinged the value of ``Module["wasmBinary"]`` - ``wasmBinary`` will then be captured in ``getBinary``, ``getBinary`` in turn gets captured in ``instantiateSync`` which ultimately gets captured in ``createWasm`` - each time it's the only reference. ``createWasm`` is then only called once - so since I guess the local functions including ``createWasm`` will fall out of scope after the module creation, it can at least be garbage-collected in theory (after changing it to resetting ``Module["wasmBinary"]`` itself at the end). No idea, if it will actually happen of course :-).

Reply:
If something stores a reference to it, must have already doing it before, right? So with the reset seems that at least we're not making the situation worse.